pkg/cmd/app/translate.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2021 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/base64"
+	"math"
+	"strings"
+
+	"github.com/okteto/okteto/pkg/k8s/configmaps"
+	"github.com/okteto/okteto/pkg/model"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	nameField     = "name"
+	statusField   = "status"
+	outputField   = "output"
+	repoField     = "repository"
+	branchField   = "branch"
+	filenameField = "filename"
+	yamlField     = "yaml"
+
+	// ProgressingStatus indicates that an app is being deployed
+	ProgressingStatus = "progressing"
+	// DeployedStatus indicates that an app is deployed
+	DeployedStatus = "deployed"
+	// ErrorStatus indicates that an app has errors
+	ErrorStatus = "error"
+	// DestroyingStatus indicates that an app is being destroyed
+	DestroyingStatus = "destroying"
+
+	// maxLogOutput is the maximum size that we allow to allocate for logs.
+	// Specifically 800kb. The limit on configmaps is 1Mb but we want to leave some
+	// room for the other data stored in there.
+	// Note that the is the limit after encoding the logs to base64 which is how
+	// the logs are stored in the configmap.
+	maxLogOutput = 800 << (10 * 1)
+)
+
+// maxLogOutputRaw is the maximum size we allow to allocate for logs before
+// being encoded to base64
+// See: https://stackoverflow.com/a/4715480/1100238
+var maxLogOutputRaw = int(math.Floor(float64(maxLogOutput)*3) / 4)
+
+//CfgData represents the data to be include in a configmap
+type CfgData struct {
+	Status     string
+	Output     string
+	Repository string
+	Branch     string
+	Filename   string
+	Manifest   []byte
+}
+
+// TranslateConfigMap translates the app into a configMap
+func TranslateConfigMap(name string, data *CfgData) *apiv1.ConfigMap {
+	return &apiv1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				model.GitDeployLabel: "true",
```

Comment:
Which label should we use? Right now we are using the same as the pipeline deploy

Reply:
right